src/Nethermind/Nethermind.Crypto/BlsSigner.cs

```diff
@@ -23,96 +23,132 @@ public static class BlsSigner
     private const int InputLength = 64;
 
     [SkipLocalsInit]
+    // buf must be of size G2.Sz
+    public static Signature Sign(Span<long> buf, Bls.SecretKey sk, ReadOnlySpan<byte> message)
+    {
+        if (buf.Length != G2.Sz)
+        {
+            throw new ArgumentException($"Signature buffer {nameof(buf)} must be of size {G2.Sz}.");
+        }
+
+        G2 p = new(buf);
+        Signature s = new(p);
+        s.Sign(sk, message);
+        return s;
+    }
+
     public static Signature Sign(Bls.SecretKey sk, ReadOnlySpan<byte> message)
+        => Sign(new long[G2.Sz], sk, message);
+
+    public static bool Verify(G1Affine publicKey, Signature signature, ReadOnlySpan<byte> message)
     {
-        G2 p = new(stackalloc long[G2.Sz]);
-        p.HashTo(message, Cryptosuite);
-        p.SignWith(sk);
-        return new(p.Compress());
+        int len = 2 * GT.Sz;
+        using ArrayPoolList<long> buf = new(len, len);
```

Comment:
would it be better to stackalloc?
Do you need stack memory cleared in this method?

Reply:
GT points are quite large so don't think it's safe to allocate on the stack

Reply:
How large?